common/v2/features/SendAssets/components/SendAssetsForm.tsx

```diff
@@ -255,6 +255,19 @@ export default function SendAssetsForm({ txConfig, onComplete }: IStepComponentP
           onComplete(fields);
         }}
         render={({ errors, setFieldValue, setFieldTouched, touched, values, handleChange }) => {
+          // Handle nonce calculation
+          useEffect(() => {
+            if (R.isEmpty(values) || R.isEmpty(values.network) || R.isEmpty(values.account)) {
+              return;
+            }
+            setIsEstimatingNonce(true);
+            (async () => {
+              const nonce: number = await getNonce(values.network, values.account.address);
+              setFieldValue('nonceField', nonce.toString());
+              setIsEstimatingNonce(false);
+            })();
+          }, [values.network, values.account]);
+
```

Comment:
Danger: According to the docs effects should only be declared at the top level function:
https://reactjs.org/docs/hooks-rules.html

Reply:
HOC is a top level function. Setting values in parent component with setState would trigger infinitive loop. But anyway this was fixed by https://github.com/MyCryptoHQ/MyCrypto/pull/3161. So I will just close this PR